dashboard/app/controllers/user_levels_controller.rb

```diff
@@ -23,9 +23,27 @@ def destroy
     head :no_content
   end
 
+  def delete_predict_level_progress
+    script = Script.get_from_cache(params[:script_id])
+    return head :not_found, text: 'Script not found' unless script
+    return head :forbidden, text: 'User must be instructor of course' unless script.can_be_instructor?(current_user)
+    level = Level.find(params[:level_id])
+    return head :not_found, text: 'Level not found' unless level
+    return head :bad_request, text: "Clearing progress on level type #{level.type} is not supported" unless ['Multi', 'FreeResponse'].include?(level.type)
+    UserLevel.where(user_id: current_user.id, script_id: script.id, level: level.id).destroy_all
```

Comment:
Two questions here:
1) What gets shown when an instructor tries to delete another person's progress? Is that a possibility in this context?
2) Does any of this make sense to have in the `ability.rb` file? @dmcavoy I know did some work in this area

Reply:
1. Because this request only has the script and level id, they can't try to delete another person's progress, only their own.

2. Definitely interested to know if some of the logic around being an instructor of the course can go into ability.rb!

Reply:
Both sound good! I think Dani knows more on (2) than I do, so I'll leave that to her